text/0000-rust-for-linux-project-goal-2024h2.md

```diff
@@ -0,0 +1,238 @@
+* **Don't let perfect be the enemy of good.** The primary goal is to offer stable support for the particular use cases that the Linux kernel requires. Wherever possible we aim to stabilize features completely, but if necessary, we can try to stabilize a subset of functionality that meets the kernel developers' needs while leaving other aspects unstable.
+
+# Ownership and other resources
+
+Here is a detailed list of the work to be done and who is expected to do it. This table includes the work to be done by owners and the work to be done by Rust teams (subject to approval by the team in an RFC/FCP).
+
+* The ![Funded][] badge indicates that the owner has committed and work will be funded by their employer or other sources.
+* The ![Team][] badge indicates a requirement where Team support is needed.
+
+| Subgoal                                              | Owner(s) or team(s)                | Status            |
+| ---------------------------------------------------- | ---------------------------------- | ----------------- |
+| overall program management                           | [nikomatsakis][], [joshtriplett][] | ![Funded][]       |
+| arbitrary self types v2                              |                                    |                   |
+| ↳ ~~author [RFC][RFC 3519]~~                         | ~~[Adrian Taylor][]~~              | ![Complete][]     |
+| ↳ ~~approve [RFC][RFC 3519]~~                        | ~~[Lang]~~                         | ![Complete][]     |
+| ↳ implementation                                     | [Adrian Taylor][]                  | ![Funded][]       |
+| ↳ assigned reviewer                                  | ![Team] [Compiler]                 | ![Not approved][] |
+| ↳ stabilization                                      | [Adrian Taylor][]                  | ![Funded][]       |
+| derive smart pointer                                 |                                    |                   |
+| ↳ ~~author [RFC][RFC 3621]~~                         | ~~[Alice Ryhl][]~~                 | ![Complete][]     |
+| ↳ approve [RFC][RFC 3621]                            | ![Team][] [Lang]                   | ![Approved][]     |
+| ↳ implementation                                     | [Xiang][]                          | ![Volunteer][]    |
+| ↳ stabilization                                      | [Xiang][]                          | ![Volunteer][]    |
+| `asm_goto`                                           |                                    |                   |
+| ↳ ~~implementation~~                                 | -                                  | ![Complete][]     |
```

Comment:
I am happy to also own the implementation part of asm_goto. In my implementation [#119365](https://github.com/rust-lang/rust/pull/119365) I leave out the fallthrough part, and I am happy to implement that if experiment in the kernel shows that it is useful.